plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSessionProperties.java

```diff
@@ -108,6 +120,16 @@ public PinotSessionProperties(PinotConfig pinotConfig)
                         "Number of segments of the same host per split",
                         pinotConfig.getSegmentsPerSplit(),
                         value -> checkArgument(value > 0, "Number of segments per split must be more than zero"),
+                        false),
+                booleanProperty(
+                        AGGREGATION_PUSHDOWN_ENABLED,
+                        "Enable aggregation pushdown",
+                        pinotConfig.isAggregationPushdownEnabled(),
+                        false),
+                booleanProperty(
+                        DISTINCT_COUNT_PUSHDOWN_ENABLED,
+                        "Enable distinct count pushdown",
+                        pinotConfig.isDistinctCountPushdownEnabled(),
```

Comment:
Add validation to ensure this cannot be set without setting `pinot.aggregation-pushdown.enabled` too.

Reply:
Without passing the session into the validator I do not see a way to do this. Instead I added the check for aggregation pushdown being enabled to `isCountDistinctPushdownEnabled`.

Reply:
@martint Is there any to way validate session properties based on other session properties? The only examples I could find do it in the getter for the session property (https://github.com/trinodb/trino/blob/d27254da35f821596d089f587be5913e6e39b7c7/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSessionProperties.java#L492).

Reply:
Thanks! I added a verify as the `PinotSessionProperties.isCountDistinctPushdownEnabled()` method would never be called unless aggregation pushdown is enabled.